apple/macbook-pro/12-1/default.nix

```diff
@@ -7,6 +7,14 @@
     <nixpkgs/nixos/modules/hardware/network/broadcom-43xx.nix>
   ];
 
+  powerManagement = {
+    # enable gradually increasing/decreasing CPU freq, rather than using "powersave", which would keep CPU freq at 0.8Ghz.
+    cpuFreqGovernor = lib.mkDefault "conservative";
+
+    powerUpCommands = ''${pkgs.kmod}/bin/modprobe brcmfmac'';
```

Comment:
Is this both required for suspend and hibernate?

```      
Commands executed when the machine powers down. That is, they're executed both when the system shuts down and when it goes to suspend or hibernation.
````

Reply:
for me it's required. if not using this config i would have to manually restart wpa_supplicant.

Reply:
I mean does `brcmfmac` also breaks suspend or just hibernate?

Reply:
ah, yeah, it does not break resume after suspend, so only required for hibernate.

Reply:
Ok. So it should be only unloaded for `hypernate-sleep.target` and "hybrid-sleep.target"? https://github.com/NixOS/nixpkgs/blob/950bb1452032e8e839ddb2af0dc85350c73025a6/nixos/modules/config/power-management.nix#L92

That would require two services, but systemd's script field should keep the boiler code needed for this low.

Reply:
yeah, only for:

* hibernate-sleep.target
* hybrid-sleep.target

you mean, we should override this block:

```nix
    systemd.services.post-resume =
      { description = "Post-Resume Actions";
        after = [ "suspend.target" "hibernate.target" "hybrid-sleep.target" ];
        script =
          ''
            /run/current-system/systemd/bin/systemctl try-restart post-resume.target
            ${cfg.resumeCommands}
            ${cfg.powerUpCommands}
          '';
        serviceConfig.Type = "oneshot";
      };
```

with only two targets?

generally, i'm not that fluent in nix, if you would provide the best solution here i would really appreciate it.

Reply:
No just create new systemd services:

```
    systemd.services.post-resume-brcmfmac = { 
        description = "Post-Resume Actions";
        after = [ "hibernate.target" "hybrid-sleep.target" ];
        script = ''
          # here goes your code
        '';
        serviceConfig.Type = "oneshot";
      };
```

Reply:
added.

Reply:
recalled another case: sometimes on lid close (sleep/suspend) brcmfmac could crash itself, so i guess the better approach would be to still use it in powerUp/Down commands of PM. unload/load driver.

if that's too hacky, let's leave it as it, and I might keep it in my local configuration for now.

Reply:
Ok. If this is the case, than we keep the original.

Reply:
added lib.mkBefore for both Up/Down commands.